text/010-wasm-pack-watch-and-serve.md

```diff
@@ -0,0 +1,230 @@
+to working group meetings, so it makes sense to advertise this RFC in TWiRaWA
+and on our @rustwasm Twitter account.
+
+# Detailed Explanation
+[detailed-explanation]: #detailed-explanation
+
+We will add two new subcommands to `wasm-pack`:
+
+1. `watch`
+2. `serve`
+
+## `wasm-pack watch`
+
+The `watch` subcommand will observe the crate directory for changes and
+automatically re-build when files change.
+
+### Command Line Interface
+
+The `watch` command takes an identical set of command line arguments as
+`wasm-pack build` does.
+
+```
+USAGE:
+    wasm-pack watch [FLAGS] [OPTIONS] [path] [-- <extra_options>...]
+
+FLAGS:
+        --dev              Create a development build. Enable debug info, and disable optimizations.
+        --no-typescript    By default a *.d.ts file is generated for the generated JS file, but this flag will disable
+                           generating this TypeScript file.
+    -h, --help             Prints help information
+        --profiling        Create a profiling build. Enable optimizations and debug info.
+        --release          Create a release build. Enable optimizations and disable debug info.
+    -V, --version          Prints version information
+
+OPTIONS:
+    -m, --mode <mode>          Sets steps to be run. [possible values: no-install, normal, force] [default: normal]
+    -d, --out-dir <out_dir>    Sets the output directory with a relative path. [default: pkg]
+    -s, --scope <scope>        The npm scope to use in package.json, if any.
+    -t, --target <target>      Sets the target environment. [possible values: browser, nodejs, no-modules] [default:
+                               browser]
+
+ARGS:
+    <path>                The path to the Rust crate.
+    <extra_options>...    List of extra options to pass to `cargo build`
+```
+
+### Implementation
+
+We can use [the `notify` crate][notify] to watch the filesystem for
+changes. Initially, we will just watch the crate directory for
+changes. Eventually, we can use [`cargo build --build-plan`][build-plan] to get
+a list of files that we should be watching.
+
+[notify]: https://crates.io/crates/notify
+[build-plan]: https://github.com/rust-lang/cargo/issues/5579
+
+## `wasm-pack serve`
+
+The `serve` subcommand starts a local Web server in the crate directory, watches
+for file changes, and re-builds the crate on changes.
+
+### Command Line Interface
+
+The `serve` subcommand takes a superset of the arguments that `wasm-pack build`
+takes. Like other `wasm-pack` subcommands, it takes an optional path to the
+crate directory as a positional argument, but if that is missing, then it
+defaults to the current directory. It has the usual flags for controlling the
+build profile, and the target environment. It takes extra options after `--`
+that get passed through straight to `cargo build`.
+
+Additionally, it has a `--no-watch` flag to disable watching the crate for
```

Comment:
Maybe this should also be able to be controlled from wasm-pack's toml configuration? https://rustwasm.github.io/wasm-pack/book/cargo-toml-configuration.html